### PR TITLE
nc-news-7 add post endpoint function + tests

### PR DESCRIPTION
**Pull request by jetakazono**

### CODE REVIEW

**Comment by jetakazono** [comment]
function selectUserByUsername will be placed in your model once it has one

File: __tests__/app.test.js [hunk 1]
```diff
@@ -184,3 +185,80 @@ describe("GET /api/articles/:article_id/comments", () => {
...
+            .expect(201)
+            .then(({ body }) => {
+                const { comment } = body
+
+                expect(typeof comment).toBe("object")
+                expect(Object.keys(comment)).toHaveLength(6)
+                expect(comment).toMatchObject({
+                    comment_id: expect.any(Number),
+                    body: expect.any(String),
+                    article_id: expect.any(Number),
```
Comment (pippaamy) [inline __tests__/app.test.js:206]: we can expect the actual article id here :)

File: __tests__/app.test.js [hunk 2]
```diff
@@ -184,3 +185,80 @@ describe("GET /api/articles/:article_id/comments", () => {
...
+            .then(({ body }) => {
+                expect(body.msg).toBe("bad request")
+            })
+    })
+    test("status: 400 - should responds with bad request when required values is not given", () => {
+        const testComment = {
+            username: "butter_bridge",
+        }
+        return request(app)
+            .post("/api/articles/NaN/comments")
```
Comment (pippaamy) [inline __tests__/app.test.js:231]: I would test this with a valid id in the path

File: __tests__/app.test.js [hunk 3]
```diff
@@ -184,3 +185,80 @@ describe("GET /api/articles/:article_id/comments", () => {
...
+            .send(testComment)
+            .expect(400)
+            .then(({ body }) => {
+                expect(body.msg).toBe("bad request")
+            })
+    })
+    test("status 404 - should responds with not found when username is an a non existent user", () => {
+        const testComment = {
+            username: "obi",
```
Comment (pippaamy) [inline __tests__/app.test.js:240]: Obi!!!!🥰🐶

File: __tests__/app.test.js [hunk 4]
```diff
@@ -184,3 +185,80 @@ describe("GET /api/articles/:article_id/comments", () => {
             })
     })
 })
+describe("POST /api/articles/:article_id/comments", () => {
+    test("status: 200 - should responds with the posted comment", () => {
```
Comment (pippaamy) [inline __tests__/app.test.js:189]: this should describe 201

File: __tests__/app.test.js [hunk 5]
```diff
@@ -184,3 +185,80 @@ describe("GET /api/articles/:article_id/comments", () => {
...
+        return request(app)
+            .post("/api/articles/11/comments")
+            .send(testComment)
+            .expect(201)
+            .then(({ body }) => {
+                const { comment } = body
+
+                expect(typeof comment).toBe("object")
+                expect(Object.keys(comment)).toHaveLength(6)
+                expect(comment).toMatchObject({
```
Comment (pippaamy) [inline __tests__/app.test.js:203]: let add another 201 that It ignores unnecessary properties

File: endpoints.json [hunk 6]
```diff
@@ -42,5 +42,39 @@
...
+          "title": "Seafood substitutions are increasing",
+          "topic": "cooking",
+          "created_at": "2018-05-30T15:59:13.341Z",
+          "votes": 0,
+          "article_img_url": "https://images.pexels.com/photos/158651/news-newsletter-newspaper-information-158651.jpeg?w=700&h=700",
+          "comment_count": 6
+        }
+      ]
+    }
+  },
```
Comment (pippaamy) [inline endpoints.json:63]: well remembered :)

File: models/articles.models.js [hunk 7]
```diff
@@ -40,3 +40,30 @@ exports.selectCommentsByArticleId = (article_id) => {
...
+            (body, author, article_id)
+        VALUES ($1, $2, $3)
+        RETURNING *;`
+
+    return db.query(queryStr, [body, username, article_id]).then(({ rows }) => {
+        return rows[0]
+    })
+}
+
+exports.selectUserByUsername = (username) => {
```
Comment (pippaamy) [inline models/articles.models.js:56]: a psql error will happen if the username isn't in the database so this functionality isn't needed 😫

**Review by pippaamy — COMMENTED** [review]
Nice work, left some changes for you to make 🤩

File: models/articles.models.js (hunk 7 above)
Comment (jetakazono) [reply models/articles.models.js:56]: should it be a 400 then?

**Pull request merged**